bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/AbstractBigtableTable.java

```diff
@@ -752,4 +710,69 @@ public void setWriteRpcTimeout(int i) {
   public int getRpcTimeout() {
     throw new UnsupportedOperationException("getRpcTimeout");
   }
+
+  private static Class<? extends AbstractBigtableTable> tableClass = null;
+
+  /**
+   * This is a workaround for incompatible changes in hbase minor versions. Dynamically generates a
+   * class that extends AbstractBigtableTable so incompatible methods won't be accessed unless the
+   * methods are called. If a method is implemented by AbstractBigtableTable, the generated class
+   * will invoke the implementation in AbstractBigtableTable.
+   */
+  private static synchronized Class<? extends AbstractBigtableTable> getSubclass()
+      throws NoSuchMethodException {
+    if (tableClass == null) {
+      tableClass =
+          new ByteBuddy()
+              .subclass(AbstractBigtableTable.class)
+              .method(ElementMatchers.named("coprocessorService"))
+              .intercept(
+                  InvocationHandlerAdapter.of(
+                      new AbstractBigtableAdmin.UnsupportedOperationsHandler()))
+              .method(ElementMatchers.named("batchCoprocessorService"))
+              .intercept(
+                  InvocationHandlerAdapter.of(
+                      new AbstractBigtableAdmin.UnsupportedOperationsHandler()))
+              .method(
+                  ElementMatchers.named("mutateRow")
+                      .and(ElementMatchers.returns(TypeDescription.VOID)))
+              .intercept(
+                  MethodCall.invoke(
+                          AbstractBigtableTable.class.getDeclaredMethod(
+                              "mutateRowVoid", RowMutations.class))
+                      .withAllArguments())
+              .method(ElementMatchers.named("mutateRow").and(ElementMatchers.returns(Result.class)))
+              .intercept(
+                  MethodCall.invoke(
+                          AbstractBigtableTable.class.getDeclaredMethod(
+                              "mutateRowResult", RowMutations.class))
+                      .withAllArguments())
+              .make()
+              .load(
+                  AbstractBigtableTable.class.getClassLoader(),
+                  ClassLoadingStrategy.Default.INJECTION)
+              .getLoaded();
+    }
+    return tableClass;
+  }
+
+  public static AbstractBigtableTable createInstance(
+      AbstractBigtableConnection bigtableConnection, HBaseRequestAdapter hbaseAdapter)
+      throws IOException {
+    try {
+      return getSubclass()
+          .getDeclaredConstructor(AbstractBigtableConnection.class, HBaseRequestAdapter.class)
+          .newInstance(bigtableConnection, hbaseAdapter);
+    } catch (InvocationTargetException e) {
+      // Unwrap and throw IOException or RuntimeException as is, and convert all other exceptions to
+      // IOException because
+      // org.apache.hadoop.hbase.client.Connection#getAdmin() only throws
```

Comment:
This comment should be updated :) -> Connection#getTable()

Reply:
thanks :D